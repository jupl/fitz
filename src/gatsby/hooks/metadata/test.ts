import {renderHook} from '@testing-library/react-hooks'

beforeAll(async () => {
  const gatsby = await import('gatsby')
  jest.spyOn(gatsby, 'useStaticQuery').mockReturnValue({
    site: {
      metadata: {
        author: 'Author',
        description: 'Description',
        title: 'Title',
      },
    },
  })
})

afterAll(async () => {
  const gatsby = await import('gatsby')
  jest.spyOn(gatsby, 'useStaticQuery').mockRestore()
})

test('hook', async () => {
  const {useGatsbyMetadata} = await import('.')
  const {result} = renderHook(() => useGatsbyMetadata())
  expect(result.current).toMatchSnapshot()
})
