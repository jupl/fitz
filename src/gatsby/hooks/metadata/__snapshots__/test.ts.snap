// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hook 1`] = `
Object {
  "author": "Author",
  "description": "Description",
  "title": "Title",
}
`;
